py4j-java/src/main/java/py4j/ClientServerConnection.java

```diff
@@ -74,6 +74,10 @@ public ClientServerConnection(Gateway gateway, Socket socket, List<Class<? exten
 					throws IOException {
 		super();
 		this.socket = socket;
+
+		// disable Nagle's algorithm on loopback connection to avoid performance issues due to write,write,read situations
+		if (socket.getLocalAddress().isLoopbackAddress()) socket.setTcpNoDelay(true);
```

Comment:
Do we have a rough evidence for this performance improvement? e.g., benchmark.

Reply:
And what's the impact of disabling this?

Reply:
See https://github.com/py4j/py4j/issues/516 for background and a measurement for a slow case. For the loopback interface I would assume, there are no negative performance impacts for disabling Nagle. I would have assumed it to be the default on loopback, but that seems not to be always the case.